src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = 8;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
 
+        if (size >= threshold) {
+            resize();
+        }
+        putValue(hash(key), key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Node<K, V> newNode = table[index];
+        while (newNode != null) {
+            if (Objects.equals(key, newNode.key)) {
+                return newNode.value;
+            }
+            newNode = newNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(Object key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private void putValue(int hash, K key, V value) {
+        if (table[hash] == null) {
+            table[hash] = new Node<>(key, value, null);
+        } else {
+            Node<K, V> newNode = table[hash];
+            while (newNode.next != null) {
+                if (Objects.equals(key, newNode.key)) {
+                    break;
+                }
+                newNode = newNode.next;
+            }
+            if (Objects.equals(key, newNode.key)) {
+                newNode.value = value;
+                size--;
+            } else {
+                newNode.next = new Node<>(key, value, null);
+            }
+        }
+    }
```

Comment:
lets remove duplicate creation of new Node and _**try**_ (it's almost ok as it is) to rewrite this method without if { } else
hint: use return, it will help to not go 3 tabs deep and increase readability

Reply:
fixed all, except : "remove duplicate creation of new Node ". I dont see how to fix this (